crates/forge/src/lib.rs

```diff
@@ -194,6 +195,61 @@ fn try_close_tmp_dir(temp_dir: TempDir) -> Result<()> {
     Ok(())
 }
 
+fn find_or_parse_raw_fork_config(
+    raw_fork_config: RawForkConfig,
+    runner_config: &RunnerConfig,
+) -> Result<ValidatedForkConfig> {
+    match raw_fork_config {
+        RawForkConfig::Params(url_str, block_id) => Ok(ValidatedForkConfig {
+            url: url_str.parse()?,
+            block_id,
+        }),
+        RawForkConfig::Id(name) => {
+            let fork_target_from_runner_config = runner_config
+                .fork_targets
+                .iter()
+                .find(|fork| fork.name == name)
+                .ok_or_else(|| {
+                    anyhow!("Fork configuration named = {name} not found in the Scarb.toml")
+                })?;
+            Ok(ValidatedForkConfig {
+                url: fork_target_from_runner_config.url.clone(),
+                block_id: fork_target_from_runner_config.block_id,
+            })
+        }
+    }
+}
+
+fn map_fork_configs_from_raw_to_validated_in_tests(
+    compiled_test_crate: CompiledTestCrate<RawForkConfig>,
+    runner_config: &RunnerConfig,
+) -> Result<CompiledTestCrate<ValidatedForkConfig>> {
+    let mut test_cases = vec![];
+
+    for case in compiled_test_crate.test_cases {
+        let fork_config = if let Some(fc) = case.fork_config {
+            Some(find_or_parse_raw_fork_config(fc, runner_config)?)
+        } else {
+            None
+        };
+
+        test_cases.push(TestCase {
+            name: case.name,
+            available_gas: case.available_gas,
+            ignored: case.ignored,
+            expected_result: case.expected_result,
+            fork_config,
+            fuzzer_config: case.fuzzer_config,
+        });
+    }
+
+    Ok(CompiledTestCrate {
+        sierra_program: compiled_test_crate.sierra_program,
+        test_cases,
+        tests_location: compiled_test_crate.tests_location,
+    })
+}
+
```

Comment:
I think this should be done somewhere in the `collecting.rs` module and treated as part of test collection or compilation. Details of fork configs should not concern us later. 

We should only be dealing with ValidatedForkConfig post that stages.